fe/fe-core/src/main/java/com/starrocks/sql/common/TypeManager.java

```diff
@@ -93,8 +93,15 @@ public static Expr addCastExpr(Expr expr, Type targetType) {
     public static Type getCompatibleTypeForBetweenAndIn(List<Type> types) {
         Preconditions.checkState(types.size() > 0);
         Type compatibleType = types.get(0);
+        // TODO(mofei) support it
+        if (compatibleType.isJsonType()) {
+            throw new SemanticException("InPredicate of JSON is not supported");
+        }
 
         for (int i = 1; i < types.size(); i++) {
+            if (types.get(i).isJsonType()) {
+                throw new SemanticException("InPredicate of JSON is not supported");
```

Comment:
Relevant semantic checking of expressions should be done in ExpressionAnalyzer, not in TypeManager

Reply:
fixed